pyteal/ast/subroutine.py

```diff
@@ -22,11 +22,12 @@ def __init__(
         implementation: Callable[..., Expr],
         returnType: TealType,
         nameStr: str = None,
+        input_types: List[TealType] = None,
```

Comment:
Adding optional input types. I needed this when generating a pyteal program from a subroutine. This is because `Btoi()` needs to be applied to external `args` which must be converted to the appropriate type of the subroutine parameter.

Reply:
@tzaffi Optional - I think the comment you've left is a worthwhile inline addition within `SubroutineDefinition`.  I can see it being helpful for future maintainers to understand _why_ `input_types` was added here.

I don't feel _too_ strongly, so feel welcomed to disregard.

Reply:
👍